common4j/src/main/com/microsoft/identity/common/java/authorities/CIAMAuthority.java

```diff
@@ -0,0 +1,96 @@
+import com.microsoft.identity.common.java.logging.Logger;
+import com.microsoft.identity.common.java.providers.microsoft.azureactivedirectory.AzureActiveDirectorySlice;
+import com.microsoft.identity.common.java.providers.microsoft.microsoftsts.MicrosoftStsOAuth2Configuration;
+import com.microsoft.identity.common.java.providers.microsoft.microsoftsts.MicrosoftStsOAuth2Strategy;
+import com.microsoft.identity.common.java.providers.oauth2.OAuth2Strategy;
+import com.microsoft.identity.common.java.providers.oauth2.OAuth2StrategyParameters;
+
+import lombok.NonNull;
+
+public class CIAMAuthority extends Authority {
+    private static transient final String TAG = CIAMAuthority.class.getSimpleName();
+
+    // Adding this flag to indicate whether or not we should load the OpenId Configuration as part of handling
+    // CIAM flows. This is currently relevant for fetching the authorization endpoint from OpenId rather than
+    // using the authority itself + adding the default authorization endpoint.
+    private final boolean CIAM_USE_OPENID_CONFIGURATION = true;
+
+    public CIAMAuthority(String authorityUrl) {
+        mAuthorityTypeString = "CIAM";
+        mAuthorityUrlString = authorityUrl;
+    }
+
+    protected MicrosoftStsOAuth2Configuration createOAuth2Configuration() {
+        final String methodName = ":createOAuth2Configuration";
+        Logger.verbose(
+                TAG + methodName,
+                "Creating OAuth2Configuration"
+        );
+        MicrosoftStsOAuth2Configuration config = new MicrosoftStsOAuth2Configuration();
+        config.setAuthorityUrl(this.getAuthorityURL());
+        config.setMultipleCloudsSupported(false);
+
+        if (mSlice != null) {
+            Logger.info(
+                    TAG + methodName,
+                    "Setting slice parameters..."
+            );
+            final AzureActiveDirectorySlice slice = new AzureActiveDirectorySlice();
+            slice.setSlice(mSlice.getSlice());
+            slice.setDataCenter(mSlice.getDataCenter());
+            config.setSlice(slice);
+        }
+
+        return config;
+    }
+
+    @Override
+    public OAuth2Strategy createOAuth2Strategy(OAuth2StrategyParameters parameters) throws ClientException {
+            MicrosoftStsOAuth2Configuration config = createOAuth2Configuration();
+            final MicrosoftStsOAuth2Strategy strategy = new MicrosoftStsOAuth2Strategy(config, parameters);
+            if (CIAM_USE_OPENID_CONFIGURATION) {
+                strategy.loadOpenIdProviderConfiguration();
+            }
+            return strategy;
+    }
+
+    /**
+     * This method takes a CIAM authority string of format "tenant.ciamlogin.com" or "https://tenant.ciamlogin.com"
+     * @param authorityNoPath authority to be transformed
+     * @return full CIAM authority with path
+     */
+    public static String getFullAuthorityUrlFromAuthorityWithoutPath(@NonNull String authorityNoPath){
```

Comment:
Does this account for the temporary CIAM domain `https://ciamlogin.com/tenant` (temporary for Build 2023)? Not sure if that's in scope anymore, but I see it mentioned in some documentation (e.g. https://microsoft-my.sharepoint-df.com/:w:/r/personal/jmprieur_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7BF07A764A-DA4A-4CE9-A2F2-D6671C174C47%7D&file=Impact%20of%20CIAM%20split%20domain%20on%20DevEx.docx&_DSL=1&action=default&mobileredirect=true&cid=2cedb99e-423c-463f-98ea-0f424b9ad51d)

Reply:
I did not test with that formatting. I haven't seen it recently in the updated documentation so i did not take it into account. Will confirm with Jean-Marc

Reply:
Confirmed with Jean-marc and Ray, `https://ciamlogin.com/tenant` is not supported, so should be good on this front